src/presentation/controller/addressLookUp/Controller.ts

```diff
@@ -86,8 +86,10 @@ class AddressLookUpController extends AbstractController {
       let postcode = "";
       if (pageType === AddressLookUpPageType.choosePrincipalPlaceOfBusinessAddress) {
         postcode = cache[this.PRINCIPAL_PLACE_OF_BUSINESS_ADDRESS_CACHE_KEY]?.postal_code;
-      } else {
+      } else if (pageType === AddressLookUpPageType.chooseRegisteredOfficeAddress) {
         postcode = cache[this.REGISTERED_OFFICE_ADDRESS_CACHE_KEY]?.postal_code;
+      } else if (pageType === AddressLookUpPageType.chooseGeneralPartnerUsualResidentialAddress) {
```

Comment:
instead of the ifs, you could have a new private method getAddressCacheKey and have a lookup/map
`const addressCacheKeys = {
    [AddressLookUpPageType.choosePrincipalPlaceOfBusinessAddress]: this.PRINCIPAL_PLACE_OF_BUSINESS_ADDRESS_CACHE_KEY,
    [AddressLookUpPageType.chooseRegisteredOfficeAddress]: this.REGISTERED_OFFICE_ADDRESS_CACHE_KEY,
    [AddressLookUpPageType.chooseGeneralPartnerUsualResidentialAddress]: this.USUAL_RESIDENTIAL_ADDRESS_CACHE_KEY,
};`

Reply:
Refactored to make use of the already existing `getCacheKey` function.